src/api/app/views/webui/workflow_runs/index.html.haml

```diff
@@ -8,7 +8,8 @@
         Filtered by: #{params[:status]&.humanize}
         %i.float-end.mt-1.fa.fa-chevron-down#workflow-runs-dropdown-trigger
       .collapse#filters
-        = render WorkflowRunFilterComponent.new(token: @token, selected_filter: @selected_filter, workflow_runs: @workflow_runs)
+        = render WorkflowRunFilterComponent.new(token: @token, selected_filter: @selected_filter,
+                                                workflow_runs: @workflow_runs.except(:limit, :offset))
```

Comment:
I don't get it, you pass all of the WorkflowRuns to the component now to count them?

Reply:
Either we let the component do the counting, or we compute it in advance in the controller and we pass the value to it

Reply:
The point of the pagination is the size of `@workflow_runs` no?

Reply:
Sure it is!
I was wrong in the understanding on the computation of the `relation.all` concatenated to the `pagination`.
I understand now that the pagination by kaminari cuts down at the query level. So with the current changes the performance is becoming an issue. I'm going to move the computation in the controller and just pass the data to the component.